ops/handshake/message.go

```diff
@@ -1,52 +1,35 @@
 package handshake
 
+import "encoding/binary"
+
+type messageID int8
+
 const (
-	keepAlive = "keep-alive"
-	choke = "choke"
-	unchoke = "unchoke"
-	interested = "interested"
-	notIntersted = "not-interested"
-	have = "have"
-	request = "request"
-	cancel = "cancel"
-	piece = "piece"
+	keepAlive    messageID = 0
+	choke        messageID = 0
+	unchoke      messageID = 1
+	interested   messageID = 2
+	notIntersted messageID = 3
+	have         messageID = 4
+	bitField     messageID = 5
+	request      messageID = 6
+	piece        messageID = 7
+	cancel       messageID = 8
 )
 
-type attrib struct {
-	prefix []byte
-	messageID byte
-	payload []byte
+type Message struct {
+	MsgID   uint8
+	Payload []byte
 }
 
-// Message is a basic message type for sending messages 
-// via tcp to peers requesting data
-type Message map[string]*attrib
-
-// return a Message map, containing attribs structs as value
-// with different attrib for different key constants
-func makeMessage() Message {
-
-	message := make(Message)
-	message[keepAlive] = &attrib{prefix: []byte{0,0,0,0}, messageID: 0}
-	message[choke] = &attrib{prefix: []byte{0,0,0,1}, messageID: 0}
-	message[unchoke] = &attrib{prefix: []byte{0,0,0,1}, messageID: 1}
-	message[interested] = &attrib{prefix: []byte{0,0,0,1}, messageID: 2}
-	message[notIntersted] = &attrib{prefix: []byte{0,0,0,1}, messageID: 3}
-	message[have] = &attrib{prefix: []byte{0,0,0,5}, messageID: 5}
-	message[request] = &attrib{prefix: []byte{0,0,1,3}, messageID:6}
-	message[piece] = &attrib{prefix: []byte{0,0,0,9}, messageID:7}
-	message[cancel] = &attrib{prefix: []byte{0,0,1,3}, messageID:8}
+// convert the message into a byteslice to be sent to peers
+// based on the key provided
+func (msg Message) serializeMessage(prefixLen messageID) []byte {
 
-	return message
+	data := make([]byte, prefixLen+4)
 
-}
+	binary.BigEndian.PutUint32(data[0:4], uint32(prefixLen))
+	data[4] = byte(prefixLen)
 
```

Comment:
The payload is missing.

Reply:
In first few cases like interested and notchoking, there isn't any payload. So currently Im just going with that. THough later I'll certainly add the option for it, as of now didn't seem the need for it. What do you think?

Reply:
Yeah, you're right, we don't need that too much right now, i just want to make sure everything is working well. All good for now.